pkg/services/pluginsintegration/plugincontext/plugincontext.go

```diff
@@ -128,6 +131,48 @@ func (p *Provider) GetWithDataSource(ctx context.Context, pluginID string, user
 	return pCtx, nil
 }
 
+func (p *Provider) PluginContextForDataSource(ctx context.Context, pluginID, name string) (backend.PluginContext, error) {
```

Comment:
It's interesting that the comment for `Get` above says "If datasourceUID is not empty string
// then PluginContext.DataSourceInstanceSettings will be resolved and appended to
// returned context." which is not true :smile: Feel free to remove if you want.

Reply:
Added to a follow up (even if this stuff is short lived) https://github.com/grafana/grafana/pull/80906